src/fty_mdns_sd_server.cc

```diff
@@ -236,6 +277,52 @@ s_poll_fty_info(fty_mdns_sd_server_t *self)
     return 0;
 }
 
+void static
+s_publish_msg_service(mlm_client_t *client, AvahiResolvedServices& avahiServices) {
+
+    for(const AvahiResolvedService avahiService : avahiServices)
+    {
+        zmsg_t *msg_service = zmsg_new();
+        std::stringstream buffer;
+        buffer << "service.hostname=" << avahiService.hostname;
+        zmsg_addstr(msg_service, buffer.str().c_str());
+        buffer.str("");
+        buffer << "service.name=" << avahiService.service.name;
+        zmsg_addstr(msg_service, buffer.str().c_str());
+        buffer.str("");
+        buffer << "service.address=" << avahiService.address;
+        zmsg_addstr(msg_service, buffer.str().c_str());
+        buffer.str("");
+        buffer << "service.port=" << avahiService.port;
+        zmsg_addstr(msg_service, buffer.str().c_str());
+        for (const auto &txt : avahiService.txt) {
+            zmsg_addstr(msg_service, txt.c_str());
+        }
```

Comment:
This is fairly verbose and copy-pasty. Using `std::stringstream` here instead of concatenating two string is not necessarily the best option.

Consider something like this instead:
```
for (const auto& data : std::vector<std::string>({
  std::string("service.hostname=")+avahiService.hostname,
  std::string("service.name=")+avahiService.service.name,
  std::string("service.address=")+avahiService.address,
  std::string("service.port=")+avahiService.port,
})) {
  zmsg_addstr(msg_service, data.c_str());
}
for (const auto &txt : avahiService.txt) {
  zmsg_addstr(msg_service, txt.c_str());
}